tutoraspects/templates/aspects/apps/superset/pythonpath/collect_dbt_lineage.py

```diff
@@ -0,0 +1,195 @@
+    tables_match = regex.findall(sql)
+    tables = [table[2] + '.' + table[4] if table[2] != '' else table[4]  # full name if with schema
+              for table in tables_match
+              if table[4] != 'unnest']  # remove false positive
+    tables = set(tables)  # remove duplicates
+
+    return list(tables)
+
+
+class YamlFormatted(ruamel.yaml.YAML):
+    def __init__(self):
+        super(YamlFormatted, self).__init__()
+        self.default_flow_style = False
+        self.allow_unicode = True
+        self.encoding = 'utf-8'
+        self.block_seq_indent = 2
+        self.indent = 4
+
+
+def get_slice_tables(slice, dbt_tables):
+    """
+    Find the tables used in a dataset.
+    """
+    name = slice.table.name
+    schema = slice.table.schema
+    dataset_key = f"{schema}.{name}"
+
+    # only add datasets that are in dashboards, optionally limit to one database
+    kind = "virtual" if slice.table.is_virtual else "table"
+    if kind == "virtual":  # built on custom sql
+        sql = slice.table.get_rendered_sql()
+        tables = get_tables_from_sql(sql)
+    else:
+        tables = [dataset_key]
+
+    logger.info(f"Found tables {tables}")
+    dbt_refs = [
+        dbt_tables[table]["ref"]
+        for table in tables
+        if table in dbt_tables
+    ]
+
+    return dbt_refs
+
+
+def get_dashboard_dict(dashboard, dbt_tables):
+    """
+    Get dashboard metadata and all dataset dbt dependencies.
+    """
+    logger.info(f"Dashboard: {dashboard.slug}")
+
+    dependencies = set()
+    for slice in dashboard.slices:
+        logger.info(slice)
+        dependencies.update(
+            get_slice_tables(slice, dbt_tables=dbt_tables)
+        )
+
+    return {
+        "name": dashboard.slug,
+        "label": dashboard.dashboard_title,
+        "type": "dashboard",
+        "description": dashboard.description or "",
+        "url": dashboard.external_url or "",
+        "depends_on": list(sorted(dependencies)),
+        "owner": {"name": ""},
+    }
+
+
+def write_exposures_yaml(exposure_dashboards):
+    exposures_yaml = ruamel.yaml.comments.CommentedSeq(exposure_dashboards)
+
+    exposures_yaml_schema = {
+        'version': 2,
+        'exposures': exposures_yaml
+    }
+
+    outfile = os.path.join(DBT_DATA_PATH, "superset_exposures.yaml")
+    logger.info(f"Writing exposures to {outfile}")
+    exposures_yaml_file = YamlFormatted()
+    with open(outfile, 'w+', encoding='utf-8') as f:
+        exposures_yaml_file.dump(exposures_yaml_schema, f)
+
+
+@click.command(
+    help="""Creates a superset_exposures.yaml file in a shared data dir so that
+    aspects_dbt can add it to the project in CI and allow dbt docs to show
+    which models are being used in Superset."""
+)
+def collect_dbt_lineage():
+    """
+    Pull SQL from datasets, compare with dbt manifest.json to find used assets.
+    """
+    dbt_tables = get_tables_from_dbt()
+
+    target_dashboards = {{SUPERSET_EMBEDDABLE_DASHBOARDS}}
```

Comment:
It occurs to me that I need to change this to include the Operator Dash as well.